compose/config/types.py

```diff
@@ -118,58 +119,96 @@ def normalize_paths_for_engine(external_path, internal_path):
     """Windows paths, c:\my\path\shiny, need to be changed to be compatible with
     the Engine. Volume paths are expected to be linux style /c/my/path/shiny/
     """
-    if not IS_WINDOWS_PLATFORM:
-        return external_path, internal_path
 
-    if external_path:
-        drive, tail = os.path.splitdrive(external_path)
+    def _normalize(path):
+        drive, tail = splitdrive(path)
 
         if drive:
-            external_path = '/' + drive.lower().rstrip(':') + tail
+            path = '/' + drive.lower().rstrip(':') + tail
+
+        return path.replace('\\', '/')
+
+    if external_path:
+        external_path = _normalize(external_path)
+
+    return external_path, _normalize(internal_path)
 
-        external_path = external_path.replace('\\', '/')
 
-    return external_path, internal_path.replace('\\', '/')
+def splitdrive(path):
+    if len(path) == 0:
+        return ('', '')
+    if path[0] in ['.', '\\', '/', '~']:
+        return ('', path)
+    return ntpath.splitdrive(path)
```

Comment:
Yes, that shouldn't be an issue in that scenario. I think it's still nice to have a `splitdrive` function that can handle those corner cases and be used elsewhere, however, but if you feel strongly about it I can remove those from the list.